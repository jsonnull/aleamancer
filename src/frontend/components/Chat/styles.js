// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  border-radius: ${props => (props.isPinned ? '0' : '5px')};
  box-shadow: ${props =>
    props.isPinned ? '' : '0 3px 5px 0 rgba(0, 0, 0, 0.2)'};

  height: ${props => (props.isPinned ? '100%' : 'auto')};
  position: ${props => (props.isPinned ? 'relative' : 'absolute')};
  top: ${props => (props.isPinned ? '0' : 'auto')};
  right: ${props => (props.isPinned ? '0' : '1rem')};
  bottom: ${props => (props.isPinned ? '0' : '1rem')};

  display: flex;
  flex-direction: column;
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading
