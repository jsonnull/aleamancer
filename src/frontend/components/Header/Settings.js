// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button square onClick={showSettings}>
      <i className="fa fa-cog" />
    </Button>
  )
}

export default Settings
