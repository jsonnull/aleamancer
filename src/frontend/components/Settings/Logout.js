// @flow
import React from 'react'
import styled from 'styled-components'
import Button from '../Button'

type Props = {
  logout: Function
}

const Container = styled.div`
  margin-top: 2rem;
  display: flex;
`

const Logout = (props: Props) => {
  return (
    <Container>
      <Button red onClick={props.logout}>
        Logout
      </Button>
    </Container>
  )
}

export default Logout
