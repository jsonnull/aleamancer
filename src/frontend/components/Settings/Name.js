// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Input from '../Input'
import editable from 'frontend/hoc/editable'

type Props = {
  name: string,
  onChange: Function
}

const Editable = editable(
  styled.div`
    position: relative;
    background: transparent;
    border: none;
    border-radius: 0;
    padding-left: 0;
    border-bottom: 1px solid ${props => props.theme.borderColor};
    color: ${props => props.theme.color};
    margin: 0 0 1rem;
    width: 100%;
    &:hover {
      cursor: pointer;
    }
  `
)

const NameInput = Input.extend`
  background: transparent;
  border: none;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  color: ${props => props.theme.color};
  margin: 0 0 1rem;
  width: 100%;
  padding: 0;
`

const Name = (props: Props) => {
  return (
    <div>
      <Label>Display Name:</Label>
      <Editable
        isHovering={true}
        value={props.name}
        input={NameInput}
        onChange={props.onChange}
      />
    </div>
  )
}

export default Name
