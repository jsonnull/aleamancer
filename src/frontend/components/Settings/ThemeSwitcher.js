// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Button from '../Button'
import type { ThemeName } from 'common/types'

const DarkButton = Button.extend`
  margin-left: 5px;
`

const ThemeButtons = styled.div`
  display: flex;
`

type Props = {
  currentTheme: ThemeName,
  changeTheme: ThemeName => any
}

const ThemeSwitcher = (props: Props) => {
  const isLightTheme = props.currentTheme == 'light'
  const isDarkTheme = props.currentTheme == 'dark'

  const useGreen = { green: true }
  const useOutline = { outline: true }

  const lightButton = isLightTheme ? useGreen : useOutline
  const darkButton = isDarkTheme ? useGreen : useOutline

  return (
    <div>
      <Label>Theme</Label>
      <ThemeButtons>
        <Button {...lightButton} onClick={() => props.changeTheme('light')}>
          Light
        </Button>
        <DarkButton {...darkButton} onClick={() => props.changeTheme('dark')}>
          Dark
        </DarkButton>
      </ThemeButtons>
    </div>
  )
}

export default ThemeSwitcher
