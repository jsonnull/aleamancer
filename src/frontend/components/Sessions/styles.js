// @flow
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${props => props.theme.background};
  padding: 0 2rem;
`

export const Heading = styled.h1`
  font-size: ${fontSize.large};
  line-height: 1;
  font-family: ${fonts.heading};
  padding: 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`
