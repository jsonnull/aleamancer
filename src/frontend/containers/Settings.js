// @flow
import { connect } from 'react-redux'
import Settings from 'frontend/components/Settings'
import { changeDisplayName, changeTheme } from 'frontend/actions'
import { HIDE_SETTINGS, PERFORM_USER_LOGOUT } from 'frontend/actions/types'
import displayNameSelector from 'frontend/selectors/displayName'
import * as themes from 'frontend/styles/themes'
import type { ThemeName } from 'common/types'
import type { Theme } from 'frontend/styles/themes'
import type { State } from 'frontend/store'

export type StateProps = {
  theme: Theme,
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  theme: themes[state.preferences.theme],
  displayName: displayNameSelector(state),
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function) => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: PERFORM_USER_LOGOUT }),
  dismissSettings: () => dispatch({ type: HIDE_SETTINGS })
})

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
