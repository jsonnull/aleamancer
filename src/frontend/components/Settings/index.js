// @flow
import React from 'react'
import Modal from '../Modal'
import type { Theme } from 'frontend/styles/themes'
import type { DispatchProps, StateProps } from 'frontend/containers/Settings'
import Logout from './Logout'
import Name from './Name'
import ThemeSwitcher from './ThemeSwitcher'

type Props = {
  theme: Theme
} & StateProps &
  DispatchProps

const Settings = (props: Props) => {
  const {
    showSettings,
    displayName,
    changeDisplayName,
    theme,
    changeTheme,
    logout,
    dismissSettings
  } = props

  return (
    <Modal show={showSettings} dismiss={dismissSettings}>
      {() => (
        <div>
          <Name name={displayName} onChange={changeDisplayName} />

          <ThemeSwitcher currentTheme={theme.name} changeTheme={changeTheme} />

          <Logout logout={logout} />
        </div>
      )}
    </Modal>
  )
}

export default Settings
