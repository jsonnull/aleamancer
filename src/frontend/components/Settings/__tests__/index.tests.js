// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Settings from '../index.js'
import { light } from 'frontend/styles/themes'

describe('Settings component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={light}>
          <Settings
            showSettings={true}
            displayName="test"
            currentUserId="test"
            changeDisplayName={() => {}}
            theme={light}
            changeTheme={() => {}}
            logout={() => {}}
            dismissSettings={() => {}}
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
