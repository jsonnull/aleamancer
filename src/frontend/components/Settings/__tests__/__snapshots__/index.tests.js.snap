// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0,0,0,0.5);
  opacity: 1;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 600px;
  margin: auto;
  border-radius: 5px;
  padding: 2rem;
  box-shadow: 0 5px 10px rgba(0,0,0,0.3);
  background-color: #f7f7f7;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c5 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #8c9b7a;
  box-shadow: inset 0 -20px 10px -10px #849273;
}

.c5:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c5:hover {
  background: #7f8d6f;
  box-shadow: inset 0 -20px 10px -10px #778468;
}

.c8 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #bf3542;
  box-shadow: inset 0 -20px 10px -10px #b4323e;
}

.c8:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c8:hover {
  background: #ae303c;
  box-shadow: inset 0 -20px 10px -10px #a22d38;
}

.c7 {
  margin-top: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  font-size: 1.3rem;
  color: #3c3c3c;
  font-weight: bold;
  display: block;
}

.c3 {
  position: relative;
  background: transparent;
  border: none;
  border-radius: 0;
  padding-left: 0;
  border-bottom: 1px solid #cdc5ba;
  color: black;
  margin: 0 0 1rem;
  width: 100%;
}

.c3:hover {
  cursor: pointer;
}

.c6 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
  box-shadow: none;
  border: 1px solid black;
  color: black;
  margin-left: 5px;
}

.c6:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c6:hover {
  box-shadow: inset 0 -20px 10px -10px #d2d2d2;
  background: transparent;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      <div>
        <label
          className="c2"
        >
          Display Name:
        </label>
        <div
          className="c3"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          test
        </div>
      </div>
      <div>
        <label
          className="c2"
        >
          Theme
        </label>
        <div
          className="c4"
        >
          <button
            className="c5"
            onClick={[Function]}
          >
            Light
          </button>
          <button
            className="c6"
            onClick={[Function]}
          >
            Dark
          </button>
        </div>
      </div>
      <div
        className="c7"
      >
        <button
          className="c8"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
